app/controllers/application_controller/miq_request_methods.rb

```diff
@@ -72,7 +72,7 @@ def pre_prov_continue
   # Pre provisioning, select a template
   def pre_prov
     if params[:button] == "cancel"
-      flash_to_session(_("Add of new %{type} Request was cancelled by the user") % {:type => session[:edit][:prov_type]})
+      add_flash(_("Add of new %{type} Request was cancelled by the user") % {:type => session[:edit][:prov_type]})
```

Comment:
Did you try to debug the `flash_to_session` method? It seems like it is calling [`add_flash`](https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/helpers/application_helper/flash.rb#L4), so I don't think it is a good solution to just unwrap the method.

Reply:
This is a bit more complicated problem than it seems.

The first flash message here is added with a presenter in `replace_right_cell()`, the other one is added by `reportDataController` when rendering the GTL part. If you apply the following patch:
```
diff --git a/app/assets/javascripts/controllers/report_data_controller.js b/app/assets/javascripts/controllers/report_data_controller.js
index 975600131..e37fd2875 100644
--- a/app/assets/javascripts/controllers/report_data_controller.js
+++ b/app/assets/javascripts/controllers/report_data_controller.js
@@ -493,7 +493,7 @@
           }
         }
         gtlData.messages && gtlData.messages.forEach(function(oneMessage) {
-          add_flash(oneMessage.msg, oneMessage.level);
+          add_flash(oneMessage.message, oneMessage.level);
         });
         // Apply gettext __() on column headers
         for (var i = 0;  i < gtlData.cols.length; i++) {
```

You should see the second flash message correctly rendered.

Obviously, we need to make sure the flash message is rendered only once, but we cannot fix it by replacing plain `flash_to_session()` with `add_flash()`, since the above is a generic routine and we don't know what else that would break.

Reply:
@mzazrivec What do you think about my suggestions above? Thank you :)

Reply:
There's https://github.com/ManageIQ/manageiq-ui-classic/pull/4789 addressing this very same issue (as far as I can tell).

Reply:
@mzazrivec I tried that PR and it fixes the bug! I think I can close this PR and mention the another one as a fix for the BZs